src/dataflow/sink/kafka.rs

```diff
@@ -60,10 +40,41 @@ pub fn kafka<G>(
     // Send new schema to registry, get back the schema id for the sink.
     // TODO(benesch): don't block the worker thread here.
     let ccsr_client = ccsr::Client::new(connector.schema_registry_url.clone());
-    match ccsr_client.publish_schema(&connector.topic, &schema.to_string()) {
+    let schema_name = format!("{}-value", connector.topic);
+    match ccsr_client.publish_schema(&schema_name, &schema.to_string()) {
         Ok(schema_id) => {
             let mut config = ClientConfig::new();
             config.set("bootstrap.servers", &connector.url.to_string());
+            let admin: AdminClient<DefaultClientContext> =
+                config.create().expect("creating admin kafka client failed");
+
+            let admin_opts = AdminOptions::new().operation_timeout(Some(Duration::from_secs(5)));
+            let new_topic = NewTopic::new(&connector.topic, 1, TopicReplication::Fixed(1));
+            let res = block_on(admin.create_topics(&[new_topic], &admin_opts));
+            match res {
+                Ok(res) => {
+                    if res.len() != 1 {
+                        error!(
```

Comment:
some of the default formatting around here is kind of hard to read -- let me know if you have any suggestions on how to better structure the logic, or hint the formatting for cargo fmt

Reply:
Early returns!

Reply:
oof -- I assumed that error! did the returning